tests/common.py

```diff
@@ -1741,13 +1742,22 @@ async def snapshot_platform(
     entity_registry: er.EntityRegistry,
     snapshot: SnapshotAssertion,
     config_entry_id: str,
+    platform: Platform | None = None,
 ) -> None:
     """Snapshot a platform."""
     entity_entries = er.async_entries_for_config_entry(entity_registry, config_entry_id)
-    assert entity_entries
-    assert (
-        len({entity_entry.domain for entity_entry in entity_entries}) == 1
-    ), "Please limit the loaded platforms to 1 platform."
+    assert entity_entries, "No entities found."
```

Comment:
This makes the test fail if there are no entities?

That sounds incorrect, as that could just as well be the expected case.

Reply:
In that case you don't use this helper to snapshot _all available entities_, when there are no entities. That could just be a simple `len(hass.states)) == 0`

Reply:
i'm with @joostlek, this helper is intended to snapshot a platform including its entity states related to the given config entry id

Reply:
Sorry I do not agree. Depending on the device used with the integration, it might be very well be an expected case. Especially with more generic integration that support numerous of devices, this logic will force one to add specific tests or logic to support the solution joost put up, instead of just being able to snapshot it.

The comment from above, can simply be reversed as well.